### ISSUE TITLE
`#local` `imports` paths being recognized in `no-unassigned-import` config

### ISSUE THREAD

**Issue by fregante** [un-ts/eslint-plugin-import-x#347]
Code:

```js
import "#app/side-effects.js"
```

Config:

```js
"import-x/no-unassigned-import": [
  "error",
  {
    allow: [
      "**/*.css",
      "**/*.scss",
      "#app/side-effects.js",
    ]
  }
]
```

Error:

```rb
  1:1  error  Imported module should be assigned  import-x/no-unassigned-import
```

https://github.com/un-ts/eslint-plugin-import-x/issues/208 reportedly added support for these paths but not here. `eslint-plugin-import` is also affected.

Workaround:

```diff
"import-x/no-unassigned-import": [
  "error",
  {
    allow: [
      "**/*.css",
      "**/*.scss",
-     "#app/side-effects.js",
+     "*app/side-effects.js",
    ]
  }
]
```

**Comment by JounQin** [comment]
`exports` is not available with [the new node resolver](https://github.com/un-ts/eslint-plugin-import-x#import-xresolver-next) or https://github.com/import-js/eslint-import-resolver-typescript yet, please provide more info including runnable reproduction.

**Comment by JounQin** [comment]
It has nothing to do with `exports`/`imports` actually, it's glob's semantic:

https://github.com/isaacs/minimatch#comparisons-to-other-fnmatchglob-implementations

> If a pattern starts with #, then it is treated as a comment, and will not match anything. Use `\#` to match a literal # at the start of a line, or set the nocomment flag to suppress this behavior.

That's being said, it's easy enough to support exact matching with `===` check, see #352 for more details.
